### PR TITLE
fix: normalize external link component

### PR DESCRIPTION
**Pull request by alfonsobries**
## Summary

https://app.clickup.com/t/u3rqze

Dependants
1. https://github.com/ArkEcosystem/marketsquare.io/pull/2207
2. https://github.com/ArkEcosystem/ark.io/pull/978

Fixes some differences between external links.

**Notes:**
1. Part of the problem is that the markdown parser removes the `span` inside the external links leaving an extra blank space so I am whitelisting that tag on the markdown parser:

Screen Shot 2021-08-12 at 13 08 31

2. Fixed a regression with long links not breaking again (fixed on the screenshot)
image

3. Removes some attributes that don't make sense like the `inline` (all links are inline) and the `allowWrap` and checked all the projects that are affected (only ark.io and msq were affected) 

## To test

Merge this on the msq PR and on the ark.io PR, recompile assets and compare the links on articles, user profile, and basically, everything that is dynamic and the links added manually like the link on the footer on ark.io or the social network links on the projects.

Ensure to review every possible link you can find (search for `<x-ark-external-link` references)
 

## Checklist

-   [x] I checked my UI changes against the design and there are no notable differences
-   [x] I checked my UI changes for any responsiveness issues
-   [x] I checked my (code) changes for obvious issues, debug statements and commented code
-   [x] I provided a screenshot of my changes to the component _(if applicable)_
-   [x] I regenerated the `icons.html` file and checked if my newly added icon is shown correctly _(if necessary)_
-   [x] I added an explanation on how to use the component to the readme _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

**Comment by leMaur** [comment]
@alfonsobries I've tested thoroughly on MSQ and ARK.io and I didn't find any issue 👍🏼

**Review by leMaur — APPROVED** [review]

File: resources/views/external-link.blade.php [hunk 1]
```diff
@@ -1,26 +1,25 @@
...
+    'class'     => 'link font-semibold inline',
+    'breakAll'  => false,
     'small'     => false,
     'noIcon'    => false,
 ])
 
 <a
     href="{{ $url }}"
-    class="{{ $class }} {{ $inline ? 'inline space-x-1' : 'flex items-center space-x-2' }} {{ $allowWrap ? '' : 'whitespace-nowrap' }}"
+    class="{{ $class }} {{ $breakAll ? 'break-all' : 'break-words' }} "
```
Comment (ItsANameToo) [inline resources/views/external-link.blade.php:12]: would make it part of the class property instead, so it has `link font-semibold inline break-words`. That way if you really need `break-all` you can pass it as class anyway without needing a special property for it

**Review by ItsANameToo — CHANGES_REQUESTED** [review]

**Comment by alfonsobries** [comment]
@ItsANameToo ✅

**Comment by ItsANameToo** [comment]
See comment on MSQ PR @alfonsobries

**Comment by alfonsobries** [comment]
@ItsANameToo mm... apparently only happens on links that have text on the right side, fixed by adding the underline to the span on external links (to prevent the underline on the icon).

✅

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**